### DISCUSSION TITLE
Contributing to this project & PR's

### DISCUSSION THREAD

**Discussion by gbowne1** [gbowne1/codebooker#119] [General]
Although I have provided a CONTRIBUTING.md file I still find some minor issues with the contribution process on this project.

Perhaps some things need more clarity or clarification in the README.md and or the CONTRIBUTING.md file(s).

The things that need some improvement in this project specifically are:

- not using GitHub features like adding the CodeBooker Project to the issue and Todo/progress status

- not linking the relevant issue to the PR by using the feature at the bottom of the right side of the PR and Issue.

- Issue and PR labeling. Contributors have not been adding more details on the right side of the GitHub issue and/or PRs which will help automatically close issues and triage issues/bugs as far as priority.

- adding issues that are outside of the scope of the project

- contributors have been adding PR's with no description message(s) showing the steps they took and or screenshots.Frontend PR's would have 2 frontend team reviewers and would need 2 reviews from frontend team

- contributors not adding screenshots of working changes or sharing browser console output and errors, warnings, etc.

- I know that there's no Jest unit tests yet but contributors should provide some feedback on if and how they tested the code and the project or one of its features.  PR's containing failing tests would not get 

- Contributors contributing fixes, features, etc in a PR that were beyond the scope of the issue they were fixing. 

I have made an issue to have the developer documentation crated #102 and updated not only to help new contributors and collaborators get started contributing to the project easily, especially if they are new to react, or new to coding, or new to open source. 

We need to add more labels, milestones, etc too, to help the project out.

I would love to hear anyone's ideas on how to improve contribution to this project especially where we are lacking in documentation that covers this.

@LOGESH-B @jchansen925

**Comment by gbowne1** [comment]
One of my thoughts going forward is to:

1) once we establish app testing like unit tests, we should require passing unit tests in order to be merged and this should be included as a screenshot to any issue or PR.

2) requiring a minimum of 2 reviews from reviewers in order for PRs to get merged. 

  - Frontend PR's would have 2 frontend team reviewers and would need 2 reviews from frontend team
  - Backend  PR's would have 2 backend team reviewers and would need 2 reviews from backtend team
  
3) require screenshots of all working changes.
4) establish logging with Winston npm package 
5) All related issues must screenshots of errors, warnings, bugs in issues and in PRs
6) Implement actions and workflows like 
7) PR's assigned to contributors will getFrontend PR's would have 2 frontend team reviewers and would need 2 reviews from frontend team

**Comment by LOGESH-B** [comment]
Server side README.md file needs to update

**Reply by gbowne1** [reply]
There is no server side README.md and the README.md is on the project root.

**Reply by LOGESH-B** [reply]
There is a README at server folder tells how to setup the backend with the test user and env

You need to mention the environment setup flow of the project to the contributors

**Reply by gbowne1** [reply]
This file? https://github.com/gbowne1/codebooker/blob/master/src/server/README.md

This would get replaced by accurate developer and API documentation which would get references in the main README.md visible on GitHub. And yes we will be documenting the set up process.  

@jchansen925 will be working on this
